code/game/objects/items/weapons/medigun.dm

```diff
@@ -0,0 +1,360 @@
+	var/healing = 0.5
+	if(emagged)
+		healing = -1
+	if(istype(M,/mob/living/carbon/human))
+		var/mob/living/carbon/human/H = M
+		var/datum/organ/internal/heart/heartcheck = H.internal_organs_by_name["heart"]
+		if(heartcheck.uber)
+			healing = 1
+
+	M.adjustOxyLoss(-2*healing)
+	M.heal_organ_damage(healing*2, 0)
+	M.heal_organ_damage(0, healing*2)
+	M.adjustToxLoss(-2*healing)
+	M.adjustCloneLoss(-2*healing)
+	M.adjustBrainLoss(-2*healing)
+	M.adjustBruteLoss(-2*healing)
+	M.adjustFireLoss(-2*healing)
+
+	if(healing > 0)
+		if(M.reagents.has_reagent("toxin"))
+			M.reagents.remove_reagent("toxin", healing)
+		if(M.reagents.has_reagent("stoxin"))
+			M.reagents.remove_reagent("stoxin", healing)
+		if(M.reagents.has_reagent("plasma"))
+			M.reagents.remove_reagent("plasma", healing)
+		if(M.reagents.has_reagent("sacid"))
+			M.reagents.remove_reagent("sacid", healing)
+		if(M.reagents.has_reagent("pacid"))
+			M.reagents.remove_reagent("pacid", healing)
+		if(M.reagents.has_reagent("cyanide"))
+			M.reagents.remove_reagent("cyanide", healing)
+		if(M.reagents.has_reagent("amatoxin"))
+			M.reagents.remove_reagent("amatoxin", healing)
+		if(M.reagents.has_reagent("chloralhydrate"))
+			M.reagents.remove_reagent("chloralhydrate", healing)
+		if(M.reagents.has_reagent("carpotoxin"))
+			M.reagents.remove_reagent("carpotoxin", healing)
+		if(M.reagents.has_reagent("mindbreaker"))
+			M.reagents.remove_reagent("mindbreaker", healing)
+
+		if(istype(M,/mob/living/simple_animal))
+			var/mob/living/simple_animal/SA = M
+			SA.health = min(SA.maxHealth,SA.health+2)
+
+		if(istype(M,/mob/living/carbon/human))
+			var/mob/living/carbon/human/H = M
+			for(var/datum/organ/external/temp in H.organs)
+				if(temp.status & ORGAN_BLEEDING)
+					temp.clamp()
+
+		if(M.losebreath >= 10)
+			M.losebreath = max(10, M.losebreath - 5)
+
+		if(M.bodytemperature > 310)
+			M.bodytemperature = max(310, M.bodytemperature - (40 * TEMPERATURE_DAMAGE_COEFFICIENT))
+		else if(M.bodytemperature < 311)
+			M.bodytemperature = min(310, M.bodytemperature + (40 * TEMPERATURE_DAMAGE_COEFFICIENT))
+
+	M.UpdateDamageIcon()
+	var/tracker_effect = "heal"
+	if(emagged)
+		tracker_effect += "red"
+	make_tracker_effects(get_turf(src), M, 1, tracker_effect, 3, /obj/effect/tracker/heal)
+	spawn(2)
+		make_tracker_effects(get_turf(src), M, 1, tracker_effect, 3, /obj/effect/tracker/heal)
+	spawn(4)
+		make_tracker_effects(get_turf(src), M, 1, tracker_effect, 3, /obj/effect/tracker/heal)
+	spawn(6)
+		make_tracker_effects(get_turf(src), M, 1, tracker_effect, 3, /obj/effect/tracker/heal)
+	spawn(8)
+		make_tracker_effects(get_turf(src), M, 1, tracker_effect, 3, /obj/effect/tracker/heal)
+	if(healing == 1)
+		if((ubercharge == 99) && istype(loc,/mob))
+			to_chat(loc,"<span class='info'>CHARGE READY</span>")
+			playsound(loc, 'sound/weapons/medigun_charged.ogg', 50, 1)
+		ubercharge = min(ubercharge+1,100)
+
+/obj/item/medigun/dropped(mob/user)
+	spawn(1)//to prevent a few exploits
```

Comment:
oh come on, I even commented that one specifically to avoid this remark.

Here it's to prevent people taking the gun away from the pack.